elm/web/rhub.py

```diff
@@ -0,0 +1,601 @@
+"""
+Code to build Corpus from the researcher hub.
+"""
+import os
+import os.path
+import logging
+from urllib.request import urlopen
+import requests
+import pandas as pd
+from bs4 import BeautifulSoup
+from rex import init_logger
+
+logger = logging.getLogger(__name__)
+init_logger(__name__, log_level='DEBUG')
+init_logger('elm', log_level='INFO')
+
+
+class ResearchOutputs():
+    """Class to handle publications portion of the NREL researcher hub."""
+    BASE_URL = "https://research-hub.nrel.gov/en/publications/?page=0"
+
+    def __init__(self, url, n_pages=1, txt_dir='./ew_txt'):
+
+        self.text_dir = txt_dir
+        self.all_links = []
+        for p in range(0, n_pages):
+            url = url + f"?page={p}"
+            with urlopen(url) as page:
+                html = page.read().decode("utf-8")
```

Comment:
Okay here's the tricky test request. Can you add a basic test that mocks the URL page read response, gives back some basic html data, and then verifies that the ResearchOutputs class and ResearcherProfiles classes can retrieve some basic info? I can help you do this but here are the basics:

1. make this page.read() call into a separate function that can be mocked
2. Run it in real life and copy the output text then save to a .txt file 
3. You can then make a dummy function in your test file that returns this cached data and then do a "mock patch" which will swap the real function for your dummy function with cached data. See example here: https://github.com/NREL/elm/blob/main/tests/test_embed.py#L36
4. Given the cached good html response, you can then run a test to make sure the classes you wrote can parse things out appropriately. 
5. Make sure you don't put any sensitive information into the cached html test data (phone numbers, emails, other stuff).